### PR TITLE
Here is my basic heroku web dyno scaler

### PR DESCRIPTION
**Pull request by stirlingolson**
It is currently being used on eLapse and IDS-NG.

I'd love any feedback.  I think the documentation is pretty good and illustrates two issues that are still remaining to be done at some point.

Thanks!

-Stirling

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,99 @@
-# heroku-scaler
+# Heroku Scaler
+
+The Heroku Scaler gem allows for free scaling of Heroku web dynos.
+It exposes 3 simple rake tasks that can be called from the free [Heroku
+Scheduler](https://elements.heroku.com/addons/scheduler).
+
+This is perfect for applications that see heavy use during known times of the day or week.  For example, we have some customers with business systems that see 99.9% of their traffic during regular business hours on the weekdays.  Since Heroku charges for dynos based on the number of seconds the dynos are running, scaling down application dynos during off-peak times can mean big savings.  Running your larger dyno setup during the 25% of normal business hours can save a bundle!  There are other tools that purport to do this but they often charge money and have, in our experience, been poorly supported.
```
Comment (artfuldodger) [inline README.md]: I'm confused by the wording of "during the 25% of normal business hours".. Would something like "Running your larger dyno setup only during business hours - which is only 25% of the time - can save a bundle!" be clearer?

File: heroku-scaler.gemspec [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  s.description = "A simple script to allow cron-based scaling of a Heroku app"
+  s.authors     = ["stirlingolson"]
+  s.email       = 'seo@foraker.com'
+  s.homepage    = 'https://github.com/foraker/heroku-scaler'
+  s.license     = 'MIT'
+
+  s.files         = `git ls-files`.split($/)
+  s.require_paths = ['lib']
+
+  s.add_dependency 'platform-api', '~> 0'
```
Comment (artfuldodger) [inline heroku-scaler.gemspec:19]: Might want to specify Rails as a dependency as well

File: lib/heroku-scaler/config.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+module HerokuScaler
+  class Config
+    attr_reader :settings
```
Comment (artfuldodger) [inline lib/heroku-scaler/config.rb]: Could make the reader private if you'd care to.

**Comment by artfuldodger** [comment]
:+1:

File: heroku-scaler.gemspec (hunk 2 above)
Comment (stirlingolson) [inline heroku-scaler.gemspec:19]: I don't think Rails has to be a dependency.  Right?  One could run this on a Sinatra Heroku app or some such?

File: README.md (hunk 1 above)
Comment (stirlingolson) [inline README.md]: Whoa.  Great catch.  That was a horrible sentence.  Thanks for the recommendation!

File: heroku-scaler.gemspec (hunk 2 above)
Comment (artfuldodger) [inline heroku-scaler.gemspec:19]: The dependency comes from the `Rails::Railtie` bit

File: lib/heroku-scaler.rb [hunk 4]
```diff
@@ -0,0 +1 @@
+require 'heroku-scaler/railtie' if defined?(Rake)
```
Comment (artfuldodger) [inline lib/heroku-scaler.rb]: You could probably have it not blow up outside of Rails if you add `&& defined?(Rails)` here, but then the Rake tasks themselves probably wouldn't be available to the including application.

File: lib/heroku-scaler/web_scale.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    def up_if_weekday!
+      up! if weekday?
+    end
+
+    def down!
+      web_scale(small_dyno_type, small_dyno_count)
+    end
+
+    def weekday?
```
Comment (jakesutton) [inline lib/heroku-scaler/web_scale.rb]: This can probably be private?

File: spec/lib/heroku-scaler/web_scale_spec.rb [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      TestFormation.new
+    end
+  end
+
+  class TestFormation
+    def update(app, type, hash)
+    end
+  end
+
+  describe WebScale do
```
Comment (jakesutton) [inline spec/lib/heroku-scaler/web_scale_spec.rb:48]: Probably want to cover `up_if_weekday!` too.

**Comment by jakesutton** [comment]
Seems alright. :+1:

File: spec/lib/heroku-scaler/web_scale_spec.rb (hunk 6 above)
Comment (stirlingolson) [inline spec/lib/heroku-scaler/web_scale_spec.rb:48]: Yeah.  Will do at some point soon.  Per the README, the weekday? business won't work in some time zones.  I'd like to fix this soon.

File: heroku-scaler.gemspec (hunk 2 above)
Comment (stirlingolson) [inline heroku-scaler.gemspec:19]: Right.  Added.  Thanks!

**Pull request merged**